testsuite/tests/typing-sigsubst/sig_local_aliases_syntax_errors.compilers.reference

```diff
@@ -0,0 +1,42 @@
+Line 6, characters 2-14:
+6 |   type t1 := A
+      ^^^^^^^^^^^^
+Error: Only type synonyms are allowed on the right of :=
+Line 3, characters 2-24:
+3 |   type t2 := { x : int }
+      ^^^^^^^^^^^^^^^^^^^^^^
+Error: Only type synonyms are allowed on the right of :=
+Line 3, characters 15-18:
+3 |   module M1 := sig end
+                   ^^^
+Error: Syntax error: 'end' expected
```

Comment:
I think you need an error production in the parser to give this a better error message